modules/afsocket/afsocket-dest.c

```diff
@@ -83,18 +83,60 @@ afsocket_dd_set_keep_alive(LogDriver *s, gboolean enable)
   self->connections_kept_alive_accross_reloads = enable;
 }
 
-static gchar *
-afsocket_dd_format_persist_name(AFSocketDestDriver *self, gboolean qfile)
+static const gchar *
+afsocket_dd_format_name(const AFSocketDestDriver* self)
+{
+  static gchar persist_name[1024];
+
+  if (self->super.super.super.persist_id)
+    {
+      g_snprintf(persist_name, sizeof(persist_name),
+                 "afsocket_dd.%s",
+                 self->super.super.super.persist_id);
+    }
+  else
+    {
+      gchar buf[MAX_SOCKADDR_STRING];
+
+      g_snprintf(persist_name, sizeof(persist_name),
+                 "afsocket_dd.(%s,%s)",
+                 (self->transport_mapper->sock_type == SOCK_STREAM) ? "stream" : "dgram",
+                 afsocket_dd_get_dest_name(self));
+    }
+
+  return persist_name;
+}
+
+static const gchar *
+afsocket_dd_format_qfile_name(const AFSocketDestDriver *self)
+{
+  static gchar persist_name[1024];
+
+  g_snprintf(persist_name, sizeof(persist_name),
+             "%s.qfile",
+             afsocket_dd_format_name(self));
+
+  return persist_name;
+}
+
+static const gchar *
+afsocket_dd_format_connections_name(const AFSocketDestDriver *self)
 {
-  static gchar persist_name[256];
+  static gchar persist_name[1024];
 
   g_snprintf(persist_name, sizeof(persist_name),
-             qfile ? "afsocket_dd_qfile(%s,%s)" : "afsocket_dd_connection(%s,%s)",
-             (self->transport_mapper->sock_type == SOCK_STREAM) ? "stream" : "dgram",
-             afsocket_dd_get_dest_name(self));
+             "%s.connections",
+             afsocket_dd_format_name(self));
+
   return persist_name;
```

Comment:
I am not sure this doesn't change the persist_name value at the end. That shouldn't be changed lightly, as these names are persisted into the persist file. And changing the name might cause current file positions to be changed across reloads.